calico/datamodel_v1.py

```diff
@@ -44,22 +44,30 @@
 POLICY_DIR = VERSION_DIR + '/policy'
 PROFILE_DIR = POLICY_DIR + "/profile"
 
+# Regex fragment that matches a valid ID such as a profile ID or an
+# endpoint ID.
+_ID_RE_FRAGMENT = r'[0-9a-zA-Z-_\.]+'
+# Validation regexes for profiles and endpoint IDs.
+GENERIC_ID_RE = re.compile(r'^' + _ID_RE_FRAGMENT + r'$')
+PROFILE_ID_RE = GENERIC_ID_RE
+ENDPOINT_ID_RE = GENERIC_ID_RE
+
 # Regex to match profile rules, capturing the profile ID in capture group
 # "profile_id".
 RULES_KEY_RE = re.compile(
```

Comment:
This means that if you have a tag that has an invalid character, instead of canonicalising it we'll just ignore it. That seems wrong to me - am I missing something?